src/components/App.js

```diff
@@ -0,0 +1,249 @@
+    const [ withdrawStater, setWithdrawStarter ] = useState( false );
+    const [ loading, setLoading ] = useState(false);
+    const [ completed, setCompleted ] = useState(false);
+    const [ enableVendorDashboard, setEnableVendorDashboard ] = useState(false);
+    const [ lastCompleted, setLastCompleted ] = useState({
+      vendor: false,
+      order: false,
+      withdraw: false
+    });
+    const [ migratable, setMigratable ] = useState(true);
+    const [ stateLoading, setStateLoading ] = useState(true);
+    const [ migrationSuccess, setMigrationSuccess ] = useState(false);
+    const [ title, setTitle ] = useState('Migrate to Dokan');
+
+    useEffect(()=>{
+      setStateLoading(true);
+      setLoading(true);
+
+      axios.post( dokan_migrator.ajax_url, null, { params: {
+        action: 'dokan_migrator_last_migrated',
+        nonce: dokan_migrator.nonce,
+      } } )
+      .then(res=>{
+        setType( res.data.data.last_migrated != 'undefined' ? res.data.data.last_migrated : 'vendor' );
+        setMigratable( res.data.data.migratable != 'undefined' ? res.data.data.migratable : false );
+        setMigrationSuccess( res.data.data.migration_success != 'undefined' ? res.data.data.migration_success : false );
+        setTitle( res.data.data.set_title != 'undefined' ? res.data.data.set_title : 'Migrate to Dokan' );
+
+        let oldData = {...lastCompleted};
+        switch (res.data.data) {
+          case 'order':
+            oldData.vendor = true;
+            break;
+
+          case 'withdraw':
+            oldData.vendor = true;
+            oldData.order = true;
+            break;
+        }
+
+        setLastCompleted(oldData);
+        setStateLoading(false);
+      });
+    },[]);
+
+    function updateMigrationState( migrated ) {
+      switch (migrated) {
+        case 'vendor':
+          setOrderStarter(true);
+          break;
+
+        case 'order':
+          setWithdrawStarter(true);
+          break;
+
+        case 'withdraw':
+          setCompleted(true);
+          setEnableVendorDashboard(true);
+          openNotification();
+          break;
+      }
+
+    }
+
+    function startMigration( start = type ) {
+      switch (start) {
+        case 'vendor':
+          setVendorStarter(true);
+          break;
+
+        case 'order':
+          setOrderStarter(true);
+          break;
+
+        case 'withdraw':
+          setWithdrawStarter(true);
+          break;
+      }
+    }
+
+    function activeVendorDashboard() {
+      axios.post( dokan_migrator.ajax_url, null, { params: {
+        action: 'dokan_migrator_active_vendor_dashboard',
+        nonce: dokan_migrator.nonce,
+      } } )
+      .then(res=>{
+        if (res.data.success) {
+          setEnableVendorDashboard(false);
+        }
+      });
+    }
+
+    const openNotification = () => {
+      notification.open({
+        message: 'Congratulations.',
+        description:
+          'You have successful migrated to Dokan. Enjoy 🎉',
```

Comment:
Don't use plain text. Kindly check all the plain texts and make them translatable using `i18n`